__tests__/app.test.js

```diff
@@ -0,0 +1,51 @@
+const request = require('supertest')
+const app = require('../app.js')
+const db = require('../db/connection.js')
+const data = require('../db/data/test-data/index.js')
+const seed = require('../db/seeds/seed.js')
+
+beforeEach(() => {
+    return seed(data)
+})
+
+afterAll(() => {
+    return db.end()
+})
+
+describe('GET /api/topics', () => {
+    it('200: returns array of topic objects with correct keys', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) => {
+            expect(Object.keys(body.topics[0]).length).toBe(2)
+            expect(Object.keys(body.topics[0])).toContain('description', 'slug')
+        })
+    });
+    it('200: returns topic array of expected length', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.topics.length).toBe(3)
+        })
+    });
+    describe('GET /api/topics ERR HANDLING', () => {
```

Comment:
This is a test for ALL bad urls. I would advise moving it into it's own separate describe block, otherwise you might end up repeating this test unnecessarily